lib/ClangImporter/ClangImporter.cpp

```diff
@@ -5657,12 +5657,17 @@ clang::FunctionDecl *ClangImporter::instantiateCXXFunctionTemplate(
     std::string failedTypesStr;
     llvm::raw_string_ostream failedTypesStrStream(failedTypesStr);
     llvm::interleaveComma(error->failedTypes, failedTypesStrStream);
+
+    std::string funcName;
+    llvm::raw_string_ostream funcNameStream(funcName);
+    func->printQualifiedName(funcNameStream);
+
     // TODO: Use the location of the apply here.
     // TODO: This error message should not reference implementation details.
     // See: https://github.com/apple/swift/pull/33053#discussion_r477003350
     ctx.Diags.diagnose(SourceLoc(),
                        diag::unable_to_convert_generic_swift_types.ID,
-                       {func->getName(), StringRef(failedTypesStr)});
+                       {StringRef(funcName), StringRef(failedTypesStr)});
```

Comment:
Is `.diagnose` going to copy when it constructs an in-flight diagnostic? Otherwise, is this going to be a dangling pointer?

I guess we do this with `failedTypesStr` too, so it's probably fine.

Reply:
Good point, we can actually avoid the creation of `StringRef`s here.

Reply:
The signature for `unable_to_convert_generic_swift_types` takes two `StringRef`s so are we just doing an implicit conversion?

Reply:
This might be safe (depending on when the diagnostic is emitted/if it's stored anywhere), I really don't know.

Reply:
Hmm, you're right. On a closer look, `StringRef`s in diagnostic parameters are either emitted in the destructor of `InFlightDiagnostic` or copied in `DiagnosticEngine::onTentativeDiagnosticFlush`, so this is safe.